### PR TITLE
sso: update go modules

### PR DESCRIPTION
**Pull request by Jusshersmith**
## Problem
Updates to certain go modules are required. This is starting to cause errors with new builds: https://circleci.com/gh/buzzfeed/sso/2185?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

## Solution

Running https://github.com/buzzfeed/sso/blob/master/scripts/test locally edited `go.mod` and `go.sum` files as required.

### CODE REVIEW

**Comment by mccutchen** [comment]
Should we also add `-mod=readonly` to the `go build` commands in our Dockerfile and Makefile to more explicitly require up-to-date module information?

**Comment by Jusshersmith** [comment]
Yeah, I think that sounds like a sensible idea @mccutchen!

File: Dockerfile [hunk 1]
```diff
@@ -12,8 +12,8 @@ ENV GO111MODULE=on
 WORKDIR /go/src/github.com/buzzfeed/sso
 
 COPY . .
-RUN cd cmd/sso-auth && go build -o /bin/sso-auth
-RUN cd cmd/sso-proxy && go build -o /bin/sso-proxy
+RUN cd cmd/sso-auth && go build --mod=readonly -o /bin/sso-auth
+RUN cd cmd/sso-proxy && go build --mod=readonly -o /bin/sso-proxy
```
Comment (mccutchen) [inline Dockerfile]: Nit, here and below:  using the single-dash form is more idiomatic for go commands: `-mod` instead of `--mod`.

Comment (Jusshersmith) [reply Dockerfile]: Oh interesting! Thanks for pointing out!

**Comment by Jusshersmith** [comment]
Think I'm going to need a fresh review on this @mccutchen! (thanks!)

**Pull request merged**